Add kraft quorum support

tinaselenge opened the pull request:
Refined configuration logic
Updated tests for quorum checks

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

tombentley committed:
rackrolling

Signed-off-by: Tom Bentley <tbentley@redhat.com>

tinaselenge committed:
Add kraft quorum support
Refined configuration logic
Updated tests for quorum checks

Check quorum health based on quorum leaders config

Stop rolling nodes in recovery state

Add test rolling pending nodes

Add tests for combined nodes.
Perform quorum check and availability check for the active controller group.

Since we have a separate group for active controller, we do not need to check active controller when batching brokers.

Address review comments

Update the logic for restarting combined and not running nodes
We now will restart all nodes in parallel if all controllers are also brokers and in not running state (e.g) so that they can form the quorum.
Made improvements to the test for checking batch of nodes restarted in parallel

Use pod label to get the node roles instead of using NodeRef
Merge contexts list and node map into a single map object for simplicity

Get the total number of controllers by counting the nodes instead of getting it from quorum state returned from Admin API.

Implement initialising of rollclient

Implement allowReconfiguration option
Add test for exceeding maxReconfig

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClient.java:
```diff
@@ -50,6 +50,14 @@ interface RollClient {
+     * @throws io.strimzi.operator.common.UncheckedExecutionException
+     * @throws io.strimzi.operator.common.UncheckedInterruptedException
+     */
+    Map<Integer, Long> describeQuorumState();
```
If this is returning the `lastCaughtUpTimestamps` the method would be better called something like `quorumLastCaughtUpTimestamps`.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClient.java:
```diff
@@ -50,6 +50,14 @@ interface RollClient {
+    Map<Integer, Long> describeQuorumState();
+
     /**
      * @return The id of the server that is the current controller of the cluster.
```
Can we change `server` -> `node`?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClient.java:
```diff
@@ -115,7 +123,7 @@ interface RollClient {
      * @throws io.strimzi.operator.common.UncheckedInterruptedException
      */
-    Map<Integer, Configs> describeBrokerConfigs(List<NodeRef> toList);
+    Map<Integer, Configs> describeKafkaConfigs(List<NodeRef> toList);
```
Should it be `describeNodeConfigs`? 

In any case you probably need to update the javadoc too, no?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClient.java:
```diff
@@ -115,7 +123,7 @@ interface RollClient {
+    Map<Integer, Configs> describeKafkaConfigs(List<NodeRef> toList);
 
-    static record Configs(Config brokerConfigs, Config brokerLoggerConfigs) { }
+    static record Configs(Config kafkaConfigs, Config nodeLoggerConfigs) { }
```
Maybe now is a good time to make this a top level, rather than inner, class?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClientImpl.java:
```diff
@@ -115,19 +119,43 @@ public List<TopicDescription> describeTopics(List<Uuid> topicIds) {
+            quorumLeader = dmqr.quorumInfo().get().leaderId();
+            return dmqr.quorumInfo().get().voters().stream().collect(Collectors.toMap(
+                    QuorumInfo.ReplicaState::replicaId,
+                    state -> state.lastCaughtUpTimestamp().isPresent() ? state.lastCaughtUpTimestamp().getAsLong() : -1));
```
I think you can use `orElse(-1)` rather than the ternary

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -318,6 +323,7 @@ void shouldRestartBrokerWithNoTopicIfReasonManualRolling() throws ExecutionExcep
                 .addNode(false, true, 0)
                 .mockHealthyNode(platformClient, rollClient, 0)
                 .mockDescribeConfigs(rollClient, Set.of(), Set.of(), 0)
+                .mockDescribeQuorum(rollClient, Map.of(0, 10000L))
```
Maybe I'm old-fashioned, but I find `10_000L` easier to read.

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -676,16 +711,69 @@ public void shouldRestartInExpectedOrder() throws ExecutionException, Interrupte
+    }
+
+    @Test
+    public void shouldRestartCombinedNodesInExpectedOrder() throws ExecutionException, InterruptedException, TimeoutException {
```
Perhaps we should have a comment on what the expected order actually is?

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -695,6 +783,10 @@ public void shouldRestartInExpectedOrderAndBatched() throws ExecutionException,
         // given
         PlatformClient platformClient = mock(PlatformClient.class);
         RollClient rollClient = mock(RollClient.class);
+        Map<Integer, Long> quorumState = new HashMap<>();
+        quorumState.put(0, 10000L);
+        quorumState.put(1, 10000L);
+        quorumState.put(2, 10000L);
```
`Map.of()` + inline

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -640,6 +670,10 @@ public void shouldRestartInExpectedOrder() throws ExecutionException, Interrupte
         // given
         PlatformClient platformClient = mock(PlatformClient.class);
         RollClient rollClient = mock(RollClient.class);
+        Map<Integer, Long> quorumState = new HashMap<>();
+        quorumState.put(0, 10000L);
+        quorumState.put(1, 10000L);
+        quorumState.put(2, 10000L);
```
`Map.of()` + inline

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -676,16 +711,69 @@ public void shouldRestartInExpectedOrder() throws ExecutionException, Interrupte
+        // given
+        PlatformClient platformClient = mock(PlatformClient.class);
+        RollClient rollClient = mock(RollClient.class);
+        Map<Integer, Long> quorumState = new HashMap<>();
+        quorumState.put(0, 10000L);
+        quorumState.put(1, 10000L);
+        quorumState.put(2, 10000L);
```
`Map.of()` + inline

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -736,15 +829,77 @@ public void shouldRestartInExpectedOrderAndBatched() throws ExecutionException,
 
         assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 1);
 
-        assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 3, 6);
+        assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 5, 8);
 
         assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 4, 7);
 
+        assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 3, 6);
```
Let's add a comment about why these expected results are like this.

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -736,15 +829,77 @@ public void shouldRestartInExpectedOrderAndBatched() throws ExecutionException,
+        // given
+        PlatformClient platformClient = mock(PlatformClient.class);
+        RollClient rollClient = mock(RollClient.class);
+        Map<Integer, Long> quorumState = new HashMap<>();
+        quorumState.put(3, 10000L);
+        quorumState.put(4, 10000L);
+        quorumState.put(5, 10000L);
+        quorumState.put(6, 10000L);
+        quorumState.put(7, 10000L);
+        quorumState.put(8, 5000L);
```
`Map.of()` + inline

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -796,12 +956,82 @@ public void shouldRestartInExpectedOrderAndBatchedWithUrp() throws ExecutionExce
+        assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 7);
+
+
+        // TODO should we fail fast (after other nodes have been restarted)
+        //  or wait for some amount of time in case those brokers not in the replicas rejoin?
```
We should decide what to do about this.

tombentley commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:
```diff
@@ -810,15 +1040,405 @@ public void shouldRestartInExpectedOrderAndBatchedWithUrp() throws ExecutionExce
         assertEquals("Cannot restart nodes {4,3} without violating some topics' min.in.sync.replicas", ex.getMessage());
 
+        //TODO: we do not restart the active controller until all the other brokers restart?
+//        assertBrokerRestarted(platformClient, rollClient, nodeRefs, rr, 6);
```
Do you still need this?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -80,6 +86,55 @@ private static boolean avail(KafkaNode kafkaNode,
         return true;
     }
 
+    private static boolean isQuorumHealthyWithoutNode(Reconciliation reconciliation, int controllerNeedRestarting, int activeControllerId, Map<Integer, Long> quorumFollowerStates) {
```
Argument-per-line

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -80,6 +86,55 @@ private static boolean avail(KafkaNode kafkaNode,
+                return false;
+            }
+        } else {
+            return numOfCaughtUpControllers >= ceil((double) (totalNumOfControllers + 1) / 2);
```
Didn't we managed to avoid `double` arithmetic last time?

tinaselenge replied:
you are right, we did. I had done this change before that.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -223,19 +278,15 @@ static List<Set<KafkaNode>> cells(Reconciliation reconciliation,
      * @return the "best" batch to be restarted
      */
-    static Set<KafkaNode> pickBestBatchForRestart(List<Set<KafkaNode>> batches, int controllerId) {
+    static Set<KafkaNode> pickBestBatchForRestart(List<Set<KafkaNode>> batches) {
```
Why did you change this?

I don't recall now if the original idea was to always put the controller in its own batch, or just make the batch with the controller be the last batch.

tinaselenge replied:
We were passing the active controlIer ID to this function and placing the batch at the end. However we put the active controller broker into `BROKER_AND_ACTIVE_CONTROLLER` group and return a separate batch for it. So I think the batches passed into this function never actually contained the active controller. 

Otherwise, we could remove `BROKER_AND_ACTIVE_CONTROLLER` group and the keep this logic to start the batch with active controller at the end. I'm happy with either.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -279,37 +332,72 @@ enum NodeFlavour {
     }
 
-    private static Set<KafkaNode> nextBatchNonActiveBrokerish(Reconciliation reconciliation,
+    private static Set<KafkaNode> nextController(Reconciliation reconciliation,
```
Can we have a little Javadoc?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -769,14 +852,13 @@ public List<Integer> loop() throws TimeoutException, InterruptedException, Execu
                 // TODO decide whether we need an explicit healthcheck here
                 //      or at least to know that the kube health check probe will have failed at the time
-                //      we break to OUTER
+                //      we break to OUTER (We need to test a scenario of breaking configuration change, does this sleep catch it?)
```
Can we do this test now?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/model/RestartReasons.java:
```diff
@@ -146,4 +146,12 @@ public int hashCode() {
+     * @param reason The reason to test.
+     * @return true if these reasons are just the single given reason.
+     */
+    public boolean singletonOf(RestartReason reason) {
```
Let's call this `isSingletonOf`, so it can't be confused with a factory method for singletons.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java:
```diff
@@ -448,25 +449,59 @@ protected Future<Void> maybeRollKafka(
-                                reconciliation,
-                                vertx,
+        if (kafka.usesKRaft()) {
+            return ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
```
We start by calling `clientSecrets` in both branches of this `if (kafka.usesKRaft())`, so can we lift that our of the `if`/push the `if` into the compose?

I'd also factor each branch into its own method: `maybeRollKafkaKraft` and `maybeRollKafkaZk`, just to keep the methods shorter.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java:
```diff
@@ -55,7 +55,15 @@ class KafkaAgentClient {
+     * @param clusterCaCertSecret Secret with the Cluster CA public key
+     * @param coKeySecret Secret with the Cluster CA private key
+     */
+    public KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
```
Parameter per line

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java:
```diff
@@ -79,7 +79,7 @@ public class KafkaBrokerConfigurationDiff extends AbstractJsonDiff {
      * @param brokerNodeRef     Broker node reference
      */
-    protected KafkaBrokerConfigurationDiff(Reconciliation reconciliation, Config brokerConfigs, String desired, KafkaVersion kafkaVersion, NodeRef brokerNodeRef) {
+    public KafkaBrokerConfigurationDiff(Reconciliation reconciliation, Config brokerConfigs, String desired, KafkaVersion kafkaVersion, NodeRef brokerNodeRef) {
```
Parameter per line

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/Time.java:
```diff
@@ -0,0 +1,159 @@
+ * <p>In practice a real system would use {@link #SYSTEM_TIME}.</p>
+ * <p>In testing you can use {@link Time.TestTime} so that tests don't depend on actually sleeping threads.</p>
+ */
+public interface Time {
```
It looks like, in Java 17+, there is now `InstantSource` in the JDK. Separately Strimzi has used the JDK's `Clock` for time abstraction where we care about time for Certificate renewals. Kate is doing some work with certificates. I don't particularly like `Clock` as an interface. We should try to figure out a single abstractions that works and use that everywhere. 

This abstraction is _similar_ to `InstantSource`, but not the same. This interface has the `sleep()` method, which abstracts `Thread.sleep`. This allows use to write  production code that uses a `Time` instance instead of `System.currentTimeMillis()`, `System.nanoTime()` and `Thread.sleep()`. Then in the tests we don't use `Time.SYSTEM_TIME`, but `TestTime`, which allows us to run the tests much quicker because the calls to `Time.sleep()` don't actually sleep the threat, but subsequent calls to `Time.systemTimeMillis()` return a time that's consistent with if it had. I would argue that this `Time` abstraction, and the associated `Alarm` class, is actually better suited to our purposes. Please could you sync with Kate and see what she needs/wants for the certificate refactor and land on something consistent. 

This and `Alarm` are quite self-contained and would be easy to pull into a separate PR and merge sooner if we decide this `Time` is the way to go.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClient.java:
```diff
@@ -0,0 +1,46 @@
+
+    /**
+     * @param nodeRef Node reference
+     * @return true if the pod for this node is not ready according to kubernetes */
```
```suggestion
     * @return true if the pod for this node is not ready according to the platform */
```

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClient.java:
```diff
@@ -0,0 +1,46 @@
+    NodeState nodeState(NodeRef nodeRef);
+
+    /**
+     * Delete the pod with the given name, thus causing the restart of the corresponding Kafka server.
```
To be a non-leaky abstraction, this interface shouldn't express its contract using Kubernetes terms.

```suggestion
     * Initiate the restart of the corresponding Kafka server.
```

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClientImpl.java:
```diff
@@ -0,0 +1,87 @@
+import java.util.Set;
+
+/**
+ * Implementation of PlatformClient
```
```suggestion
 * Implementation of PlatformClient in terms of Kubernetes Pods
```

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClient.java:
```diff
@@ -0,0 +1,46 @@
+
+    /**
+     * @param nodeRef Node reference
+     * @return Kafka process roles for this node */
```
```suggestion
     * @return Kafka process roles for this node according to the platform. This could differ from the roles that the running process actually has (for instance if the process needs to be restarted to pick up its current roles). */
```

tinaselenge replied:
Does this mean the label on the pod can be different than the actual process role?

tombentley replied:
Yes, I think so, at least in terms of the guarantees that Kube provides: If you change the labels on a pod that doesn't change what that pod is doing. _Maybe_ how the CO uses those APIs might provide some stronger guarantee, but I'm not sure off the top of my head that that is indeed the case.

We should document it either way, because it's easy to get confused about what method like this is really telling you, and from there end up with flawed logic.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClientImpl.java:
```diff
@@ -60,13 +100,49 @@ private static <T> Set<List<T>> batch(List<T> items, int maxBatchSize) {
+            // }
+            return adminClientProvider.createAdminClient(bootstrapHostnames, clusterCaCertSecret, coKeySecret, "cluster-operator");
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Failed to create admin client for brokers", e.getCause());
```
It might not be for brokers, you call this for the controller case too.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClientImpl.java:
```diff
@@ -26,20 +39,47 @@
+    private Admin brokerAdmin = null;
 
-    /** Return a future that completes when all of the given futures complete */
+    private Admin controllerAdmin = null;
```
You initialize these, but AFAICS you never close them.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RollClient.java:
```diff
@@ -21,6 +20,18 @@
+    /**
+     * Sets admin client for brokers.
+     **/
+    void initialiseBrokerAdmin();
+
+    /**
+     * Sets admin client for controllers.
+     **/
+    void initialiseControllerAdmin();
```
Now that you've implemented more of this `RollClient` it's apparent that it's more stateful than we'd like. The methods which require the agent get called early in the `loop()`, and only later do we spin up the `Admin` clients. Also the Admin clients should be closed, but it appears that the agent client doesn't need/support that. So the lifecycle is a bit messy. It might be better to separate these into two interfaces, one for the agent and another for the Admin clients.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -33,11 +37,16 @@
 
-class RackRolling {
+/**
+ * RackRolling
```
We should come up with a better name.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/NodeRoles.java:
```diff
@@ -0,0 +1,14 @@
+package io.strimzi.operator.cluster.operator.resource.rolling;
+
+/**
+ * Holds process roles for node
```
Can we say how the information here differs from the broker/controller information in a `NodeRef`?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -249,67 +315,124 @@ static Set<KafkaNode> pickBestBatchForRestart(List<Set<KafkaNode>> batches, int
-                                            Map<Integer, NodeRef> nodesNeedingRestart,
+                                            Map<Integer, Context> contextMap,
+                                            Map<Integer, NodeRoles> nodesNeedingRestart,
+                                            int totalNumOfControllers,
```
Isn't `controllerCount` slightly nicer?

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -249,67 +315,124 @@ static Set<KafkaNode> pickBestBatchForRestart(List<Set<KafkaNode>> batches, int
 
-        int controllerId = rollClient.activeController();
+        Map<Integer, Long> quorumState = rollClient.quorumLastCaughtUpTimestamps();
+        int activeControllerId = rollClient.activeController();
```
If this can be `< 0` (and the code below suggests it can) then we should update the Javadoc of the rollClient to say what that means/when it is `< 0`.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRolling.java:
```diff
@@ -249,67 +315,124 @@ static Set<KafkaNode> pickBestBatchForRestart(List<Set<KafkaNode>> batches, int
+        LOGGER.debugCr(reconciliation, "Partitioned nodes: {}", partitioned);
+
+
+        if (activeControllerId > -1) {
```
`>= 0`

tinaselenge committed:
Support remaining logs and segments in BrokerState for logging
Use consistent nodeRef in logs

Fix checkstyle complaints and tidy.
Retry the nodes when there is a TimeoutException. Re-throw TimeoutException only when max attempts is reached.

Address comments

Add unit tests for not being elected as preferred leaders
Tidy up unit tests and logics for node with NOT_RUNNING

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/AdminClient.java:
```diff
@@ -17,12 +16,30 @@
+ * An abstraction over a Kafka Admin client.
  */
-interface RollClient {
+interface AdminClient {
```
Did you mean to rename this? I think it's confusing if Kafka has an `Admin` client and we have an `AdminClient`.

tinaselenge replied:
I have reverted back this renaming. We now have a separate abstract class for KafkaAgent, which is called AgentClient but I think this might be also confusing because we have `KafkaAgentClient` already.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/AdminClient.java:
```diff
@@ -17,12 +16,30 @@
+     **/
+    void closeBrokerAdminClient();
+
+    boolean cannotConnectToNode(NodeRef nodeRef, boolean controller);
```
This one needs some javadoc

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClient.java:
```diff
@@ -25,12 +29,22 @@ enum NodeState {
+     * @param nodeRef Node reference
+     * @return Kafka process roles for this node according to the platform.
+     * This could differ from the roles that the running process actually has (for instance if the process needs to be restarted to pick up its current roles).
+     * */
```
```suggestion
     */
```

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClient.java:
```diff
@@ -25,12 +29,22 @@ enum NodeState {
-    /** @return true if the pod for this node is not ready according to kubernetes */
+    /**
+     * @param nodeRef Node reference
+     * @return NodeState according to the platform */
```
```suggestion
     * @return NodeState according to the platform 
     */
```

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/PlatformClientImpl.java:
```diff
@@ -64,8 +73,21 @@ private static boolean pendingAndUnschedulable(Pod pod) {
 
     @Override
-    public void restartNode(NodeRef nodeRef) {
+    public void restartNode(NodeRef nodeRef, RestartReasons reason) {
```
This should be `reasons` plural, I think.

tombentley commented on cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/rolling/UnrestartableNodesException.java:
```diff
@@ -4,7 +4,7 @@
 package io.strimzi.operator.cluster.operator.resource.rolling;
 
-public class UnrestartableNodesException extends RuntimeException {
+class UnrestartableNodesException extends RuntimeException {
```
Can we document this? It's the sort of thing people might see in the logs and want to understand better what it means.

tombentley reviewed (commented):
A few more minor comments.

ShubhamRwt committed:
Run the roller via single thread Executor

Added Tom's suggestions

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

tinaselenge committed:
Pass total number of controllers from the KafkaReconcilor

Add wait for unready nodes before considering them for rolling
Add wait and retry for unready nodes that don't need to be restarted
This change makes it consistent with how the current roller handles unready nodes

Fix the issue of not waiting for a node to get deleted before checking the next node
Do preferred leader election after restarting a node in RESTART_FIRST group if it's a broker node
Improve unit tests

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

ShubhamRwt committed:
Added changes w.r.t proposal and moved batching into seperate class

Fixed shouldRestartCombinedNodesInExpectedOrderAndBatchedWithUrp
Moved nodeIdToKafkaNode from Batching to Availibility class for just improvement

Added changes w.r.t to the secrets changes

Added javadocs and fixed checkstyle issues

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

ShubhamRwt committed:
Refactor and tidy up
Rebased

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

tinaselenge committed:
Add new config for max batch size
Fix unit tests

im-konge committed:
[ST] Add 0.44.0 to upgrade/downgrade tests (#10783)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

tinaselenge committed:
Remove unused userTaskID paramater from requestRebalance method in KafkaRebalanceAssemblyOperator (#10774)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

aswinayyolath committed:
fix: Enhance IPv6 Normalization Regex in IpAndDnsValidation (#10763)

Signed-off-by: Aswin A <aswin6303@gmail.com>

scholzj committed:
Add support for Kafka 3.8.1 (#10777)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Removed logback exclusions from CC (#10779)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

im-konge committed:
[ST] Add CollectorElement to the NamespaceManager for situations when we are creating Namespaces manually (#10780)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

ShubhamRwt committed:
Ability to move data between JBOD disks using Cruise Control (#10644)

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

kyguy committed:
Delete stale Cruise Control API users (#10787)

Signed-off-by: Kyle Liberti <kliberti@redhat.com>

scholzj committed:
Wait for readiness of newly created Pods before passing them to KafkaRoller (#10746)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] Fix default storage class retrieval  (#10795)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

henryZrncik committed:
[ST] fix usages of logPodImages in UpgradeSTs (#10796)

Signed-off-by: hzrncik <hzrncik@redhat.com>

PaulRMellor committed:
docs(review): minor edits to the docs from review (#10804)

Signed-off-by: prmellor <pmellor@redhat.com>

tinaselenge committed:
Handle the failure due to reaching the servlet capacity when getting user tasks (#10768)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

katheris committed:
Add tests to CaReconciler to verify trust rolling behaviour (#10784)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

ppatierno committed:
Bump upload artifact action to 4.4.3 to do deprecation of current 3.x (#10806)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Add support for Kafka 3.9.0 (#10812)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(quota plugins): note on quotas for internal users (#10803)

Signed-off-by: prmellor <pmellor@redhat.com>

venkatesh2090 committed:
Add allocateLoadBalancerNodePorts configuration to listeners configuration (#10792)

Signed-off-by: Venkatesh Kannan <venkatesprasad512@gmail.com>

Frawless committed:
Prepare upgrade files in packit pipelines in test phase where all envs are set properly (#10819)

Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

sinbadonline committed:
Add Alauda to adopter list (#10825)

Signed-off-by: Xiaolu Zhao <xlzhao@alauda.io>

tinaselenge committed:
Add tests for stopping inprogress execution when renew annotation applied (#10805)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

dependabot[bot] committed:
Bump io.netty:netty-common from 4.1.111.Final to 4.1.115.Final (#10827)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ppatierno committed:
Bumped Vert.x 4.5.11 (#10829)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

tinaselenge committed:
Replace Vert.x in CruiseControlAPI with native HttpClient API from Java (#10791)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

im-konge committed:
[ST] Remove SecurityST#testCertificates (#10830)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

PaulRMellor committed:
docs(refactor): updates the cruise control rebalance concepts (#10810)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Add/Update warnings for Zoo and MM1 removal (#10828)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] make Recovery tests run only on KRaft mode only (#10637)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

see-quick committed:
[system test] Add wait to prevent race condition in testAutoKafkaRebalanceScaleUpSc… (#10833)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

scholzj committed:
Improve logging of altering or resetting connecotr offsets (#10840)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Fix Route deletion after route listener is removed (#10845)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add warning about removal of Strimzi `EnvVarConfigProvider` and `IdentityRepicationPolicy` to `CHANGELOG.md` (#10850)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add component label to `KafkaNodePool` CRD in Helm Chart (#10854)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
Improve release_files_check script (#10856)

Signed-off-by: see-quick <maros.orsak159@gmail.com>
Signed-off-by: Maros Orsak <maros.orsak159@gmail.com>
Co-authored-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
[DOC] Clarify that Node Ports need `ClusterRoleBinding` privileges (#10857)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

katheris committed:
Only roll pods once for ClientsCa cert renewal (#10814)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

katheris committed:
Simplify Ca and CaReconciler logic (#10809)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

scholzj committed:
Remove forgotten Kafka 3.7 3rd party libs (#10862)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Improve the unit test of MM2 using the same Secret for everything (#10871)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Enable additional Kaniko options for custom container registry certificates (#10873)

Signed-off-by: Jakub Scholz <www@scholzj.com>

fvaleri committed:
Remove Kafka Connect rest.port configuration (#10874)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

im-konge committed:
[ST] Use KafkaAdmin client for checking the configured rack on Kafka node (#10869)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

katheris committed:
Annotate StrimziPodSets before rolling during CA key replacement (#10876)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

PaulRMellor committed:
docs(controllers): adds a section on controllers quorums and limitations (#10864)

Signed-off-by: prmellor <pmellor@redhat.com>

see-quick committed:
Bump up Strimzi test container 0.109.0 (#10863)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

scholzj committed:
[DOC] Remove JBOD storage limitation from KRaft migration docs (#10879)

Signed-off-by: Jakub Scholz <www@scholzj.com>

dandotimujahid committed:
Fix Unit Tests on s390x.  (#10882)

Signed-off-by: Mujahid-Dandoti <Mujahid.Dandoti@ibm.com>

fvaleri committed:
Minor: remove duplicated initial delay in KC health check config (#10881)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

scholzj committed:
[DOC] Improve the hostname verification docs for NodePort listeners (#10880)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

PaulRMellor committed:
docs(jbod): using cc to reassign partitions on jbod disks (#10837)

Signed-off-by: prmellor <pmellor@redhat.com>

ShubhamRwt committed:
Fix faulty Cruise control healthcheck  (#10884)

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

tinaselenge committed:
Allow KafkaRoller talk to controller directly (#10016)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

fvaleri committed:
Improve min.insync.replicas parsing in warnTooLargeMinIsr (#10844)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

ppatierno committed:
Update bridge to 0.31.0 (#10868)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

tinaselenge committed:
Replace use of JsonObject with JsonNode for CruiseControlApi (#10860)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

scholzj committed:
Refactor use of Informers in UO and SPS controller (#10885)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove unused fields from `@JsonPropertyOrder` annotation (#10890)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add support for CSI volumes in templates (#10891)

Signed-off-by: Jakub Scholz <www@scholzj.com>

fvaleri committed:
Add thread dump collection to report.sh (#10895)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

ppatierno committed:
Update bridge to 0.31.1 (#10896)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Fix TO logging configuration when deployed by the CO (#10899)

Signed-off-by: Jakub Scholz <www@scholzj.com>

MikeEdgar committed:
Fix Crds.kafkaTopic to include status subresource (#10906)

Signed-off-by: Michael Edgar <medgar@redhat.com>

im-konge committed:
[ST] Add STs for Connector offset management feature (#10904)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

im-konge committed:
[ST] Add check for passing the ConfigMap field to env variable using Connect Container template (#10909)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Update Helm chart README before the 0.45.0 release (#10912)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add content type to the operator metric endpoints (#10908)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Minor refactoring of how the Kubernetes Client is created and used in the Topic Operator (#10901)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] Add CruiseControl removeDisks test case (#10910)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

ppatierno committed:
Bump the main branch to Strimzi 0.46.0-SNAPSHOT (#10920)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Update Prometheus JMX Exporter to 1.1.0 (#10917)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Update Fabric8 Kubernetes Client to 7.0.0 (#10907)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Stop including MirrorMaker2 Extensions and EnvVarConfigProvider (#10923)

Signed-off-by: Jakub Scholz <www@scholzj.com>

katheris committed:
Remove Secret from ClusterCa generate cert methods (#10915)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

katheris committed:
Remove unused certs method in Ca.java (#10927)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

tinaselenge committed:
Remove Vertx from KafkaConnect API (#10911)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

im-konge committed:
[ST] Fix Javadoc warnings in STs about wrong links (#10934)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

see-quick committed:
[system test] [doc] LoggingChangeST (#10842)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

scholzj committed:
Remove MirrorMaker 1 support (#10924)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Fix network policies to allow CO talk to the controller nodes (#10940)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(edit): removes some mm1 content (#10945)

Signed-off-by: prmellor <pmellor@redhat.com>

im-konge committed:
[ST] Update mechanism for changing the Docker registry, repository, and tag - main branch (#10947)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Avoid failing the CI pipelines on "special" commit messages (#10956)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Rename `04A-Crd-kafkanodepool.yaml` to `045-Crd-kafkanodepool.yaml` (#10953)

ppatierno committed:
Added Kate Stanley as new maintainer (#10960)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Remove Connect and MM2 upgrade from Deployment support (#10930)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

ppatierno committed:
Add the 0.45.0 release to the main branch (#10962)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Remove Kafka and ZooKeeper upgrades from StatefulSets (#10935)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

scholzj committed:
Update default logging configurations for Kafka brokers and MM2 (#10961)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] Add Strimzi 0.45.0 to the upgrade/downgrade tests (#10963)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Update Fabric8 Kubernetes client to 7.0.1 (#10965)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove unused options from UO and TO logging test configurations (#10966)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] Specify correct registry, org, and tag for Helm and remove unused code for OCP (#10955)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

tinaselenge committed:
wip

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

PaulRMellor committed:
docs(KafkaNodePool): clean up feature gate description (#10974)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Remove ZooKeeper support - Part I (#10964)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] Remove un-necessary vertx Description annotation from t… (#10976)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

see-quick committed:
Update Strimzi Test Container 0.109.1 (#10914)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

scholzj committed:
Fix OPA Integration systemtests (#10980)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Deprecate ZooKeeper-based cluster fields in the API (#10979)

Signed-off-by: Jakub Scholz <www@scholzj.com>

gertschouten committed:
Exposing dnsPolicy and dnsConfig on the Strimzi api PodTemplate to allow custom name resolution for Pod resources (#10949)

Signed-off-by: gertschouten <58491508+gertschouten@users.noreply.github.com>
Co-authored-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove ZooKeeper from packaging files (#10982)

Signed-off-by: Jakub Scholz <www@scholzj.com>

showuon committed:
bump tiered storage plugin version (#10981)

Signed-off-by: Luke Chen <showuon@gmail.com>

scholzj committed:
Clean some ZooKeeper leftovers from `operator-common` module (#10984)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Clean-up the `KRaftUtils` class (#10987)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove deprecated EO Secret methods (#10993)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Consolidate methods updating `Kafka` CR status in `KafkaReconciler` (#10995)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Clean-up the `KafkaClusterCreator` class after ZooKeeper removal (#10997)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Clean `KafkaBrokerConfigurationBuilder` class after ZooKeeper removal (#10998)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Updated release documentation (#10999)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Update Helm Chart README with Zoo and MM1 removal and upgrade info (#11001)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(bug): code example formatting fix (#11003)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Remove ZooKeeper from the docs (#10988)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

scholzj committed:
Remove outdated upgrade docs (#11006)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(oauth): updates example client config for truststore (#10969)

Signed-off-by: prmellor <pmellor@redhat.com>

PaulRMellor committed:
docs(mm2): considerations for active/passive disaster recovery (#10942)

Signed-off-by: prmellor <pmellor@redhat.com>

ppatierno committed:
Fixed check Optional not null (#11015)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Fix naming of bootstrap service for oddly named listener (#11016)

Signed-off-by: Jakub Scholz <www@scholzj.com>

fvaleri committed:
Make TopicOperatorConfig consistent with the other operators (#11017)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

scholzj committed:
Fix additional volumes in MirrorMaker 2 (#11022)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] ZK removal from STs - regression tests and migration (#11018)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Remove the external service name method from `KafkaResources` (#11023)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Update Maven Builder image to 1.21 (#11024)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] [doc] Add LogSettingST  (#10938)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

im-konge committed:
[ST] Remove ZooKeeper from upgrade/downgrade tests (#11026)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

ItaiFonio committed:
Fix minor typo (#11043)

Signed-off-by: ItaiFonio <itai.fonio@sap.com>

scholzj committed:
Remove last ZooKeeper methods from KafkaResources (#11040)

Signed-off-by: Jakub Scholz <www@scholzj.com>

katheris committed:
Store Kafka node certificates in separate Secrets (#10967)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

scholzj committed:
Remove support for storage class overrides (#11034)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove unused logger from Docs and test configs - Closes #11047 (#11048)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] Remove rest of the Zookeeper occurences from STs, together with other unused code (UTO, NodePools/KRaft checks) (#11041)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

tinaselenge committed:
Handle response with status 409 when talking to KafkaConnect API (#11042)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

see-quick committed:
[system test] - Make it possible to use POSTGRES_IMAGE env  (#11044)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

scholzj committed:
Add support for CEL validation to our CRDs (#11068)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Loosen the restrictions for TLS options in `type: custom` listener authentication to enable new features (#11052)

Signed-off-by: Jakub Scholz <www@scholzj.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

ppatierno committed:
Moved bridge configuration setup within the operator (#11032)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

PaulRMellor committed:
docs(kraft): cleans up the docs for out-of-date descriptions relating to KRaft (#11071)

Signed-off-by: prmellor <pmellor@redhat.com>

ppatierno committed:
Refactoring around bridge ConfigMap naming and volume mounts (#11075)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

ppatierno committed:
Bumped Vert.x 4.5.12 and Netty 4.1.117.Final (#11076)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

ppatierno committed:
Renamed constants for config providers env vars (#11088)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

ppatierno committed:
Removed usage of `statefulset.kubernetes.io/pod-name` label on services and pods (#11087)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
[DOC] Clarify the role of the PKCS12 file in the CA secrets and remove the OpenSSL command (#11089)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(cleanup): removes or replaces content related to zookeeper or mirrormaker1 (#11093)

Signed-off-by: prmellor <pmellor@redhat.com>

see-quick committed:
Try fix race condition inside TopicControllerIT and KafkaHandlerIT (#11070)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

fvaleri committed:
Improve Cruise Control client creation and testing (#10918)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

see-quick committed:
[system test] [doc] Metrics package (#11036)

Signed-off-by: see-quick <maros.orsak159@gmail.com>
Signed-off-by: Maros Orsak <maros.orsak159@gmail.com>
Co-authored-by: henryZrncik <henrich.zrnco@gmail.com>

scholzj committed:
Improve memory consumption on large Kubernetes clusters when NodePort listeners are used (#11098)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Update Fabric8 Kubernetes Client to 7.1.0 (#11102)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Refactoring to use mapEmpty() to map to a Future<Void> (#11105)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

im-konge committed:
[ST] Update test-clients to 0.10.0 after the release (#11090)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Deprecate Secrets field in custom server authentication (#11108)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add Kafka Access Operator 0.1.1 to the packaging directory (#11112)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Prepare for Vert.x 5 (#11107)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Minor CRD Generator cleanup (#11110)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] Configure number of messages per transaction into one more test (#11111)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Avoid adding futures to immutable list (#11099)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(feature gates): updates the feature gates content (#11097)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Fix ACL Deny rules documentation (#11119)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(acls): describes handling of special users (#11118)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Update Vert.x to 4.5.13 (#11124)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Added dev documentation about running CVEs respin (#11128)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

scholzj committed:
Update KindContainer used by MockKube3 (#11129)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Trivial typo on the Helm Charts release doc (#11140)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

mstruk committed:
Remove past workaround of json-path CVE (#11135)

Signed-off-by: Marko Strukelj <marko.strukelj@gmail.com>

tinaselenge committed:
Revert PR 10016 Allow KafkaRoller talk to controller directly (#11132)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

ppatierno committed:
Fix release doc typo about helm-acceptance pipeline (#11142)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

PaulRMellor committed:
docs(tuning): updates description of managed broker configurations (#11138)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Remove the Desired Kafka replicas column from the Kafka CRD (#11137)

Signed-off-by: Jakub Scholz <www@scholzj.com>

PaulRMellor committed:
docs(api): updates description of config exclusions (#11139)

Signed-off-by: prmellor <pmellor@redhat.com>

tinaselenge committed:
Move connect configuration set up to the operator (#11062)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

PaulRMellor committed:
docs(downgrade): highlights the standard order of downgrade (#11144)

Signed-off-by: prmellor <pmellor@redhat.com>

scholzj committed:
Update Kafka Exporter to a new version 1.9.0 (#11152)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Fix deprecated GitHub actions/cache plugin (#11158)

Signed-off-by: Jakub Scholz <www@scholzj.com>

HirokiSakonju committed:
docs: fix missing escape characters (#11164)

Signed-off-by: hiroki.sakonju <hiroki.sakonju@broadleaf.co.jp>
Co-authored-by: hiroki.sakonju <hiroki.sakonju@broadleaf.co.jp>

sebastiangaiser committed:
feat(examples,metrics,kube-state-metrics): add configmap and promethe… (#10919)

Signed-off-by: Sebastian Gaiser <sebastiangaiser@users.noreply.github.com>
Signed-off-by: Sebastian Gaiser <sebastian.gaiser@hetzner-cloud.de>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

im-konge committed:
[ST] Use correct variable types for metrics, fix wrong logging (#11160)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

see-quick committed:
[system test] [perf] Add documentation to our performance tests (#11165)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

tinaselenge committed:
Remove Zookeeper relevant logic from KafkaRoller (#11149)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

see-quick committed:
[system test] [perf] Add to TO scalability tests also printing the table (#11157)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

ppatierno committed:
Added development doc about the Operators Catalog release (#11155)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

see-quick committed:
[system test] Kafka rolling update with dedicated controllers (#11031)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

im-konge committed:
[ST] Fix issue with specified Bridge image when running tests with Helm installation type and add info about configuring images into TESTING.md (#11161)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Update KindContainer to 1.4.9 and to Kube 1.32 (#11173)

Signed-off-by: Jakub Scholz <www@scholzj.com>

ppatierno committed:
Refactoring to use enhanced switch (#11174)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

MichaelMorrisEst committed:
Enable multi version single step downgrade for KRaft based clusters (#10929)

Signed-off-by: MichaelMorris <michael.morris@est.tech>
Signed-off-by: Michael Morris <105736419+MichaelMorrisEst@users.noreply.github.com>
Co-authored-by: Paolo Patierno <paolo.patierno@gmail.com>

scholzj committed:
Remove unused fields from `kafka-versions.yaml` file (#11175)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Remove the Log Message and Inter Broker Protocol annotations (#11177)

Signed-off-by: Jakub Scholz <www@scholzj.com>

im-konge committed:
[ST] Change the way how we assert values in logs from Pods (#11187)

see-quick committed:
Add Podman Compatibility and Enhance Kind cluster setup (#11086)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

sebastiangaiser committed:
feat(examples,metrics,kube-state-metrics): extend configmap and prome… (#11189)

Signed-off-by: Sebastian Gaiser <sebastian.gaiser@hetzner-cloud.de>
Signed-off-by: Sebastian Gaiser <sebastiangaiser@users.noreply.github.com>
Co-authored-by: Sebastian Gaiser <sebastian.gaiser@hetzner-cloud.de>
Co-authored-by: Paolo Patierno <paolo.patierno@gmail.com>

PaulRMellor committed:
docs(cruise control): updates limitations note for cruise control (#11188)

Signed-off-by: prmellor <pmellor@redhat.com>

see-quick committed:
Remove JenkinsFile for x64 arch (#11196)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

see-quick committed:
Load also ip_tables kernel module into podman VMs, Also fix pasta issue. (#11193)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

im-konge committed:
[ST] Fix creation of CollectorElement based on installation type (#11197)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

scholzj committed:
Fix Mirror Maker 2 connector reconciliation race condition (#11194)

Signed-off-by: Jakub Scholz <www@scholzj.com>

scholzj committed:
Add support for Kafka 4.0.0 (#11176)

Signed-off-by: Jakub Scholz <www@scholzj.com>

pnagy-cldr committed:
[ST] Annotate the Pod resources instead of the StrimziPodSet resource for starting a rolling restart in KafkaRollerST (#11204)

Signed-off-by: Patrik Nagy <pnagy@cloudera.com>

ppatierno committed:
Refactoring with enhanced switch (#11211)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

im-konge committed:
[ST] Add Kafka Access Operator ST (#11192)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

PaulRMellor committed:
docs(operators): updates to the operator overviews (#11163)

Signed-off-by: prmellor <pmellor@redhat.com>

urbandan committed:
Add Cloudera to ADOPTERS file (#11213)

Signed-off-by: Daniel Urban <urb.daniel7@gmail.com>

scholzj committed:
Deprecate the `type: opa` authorizer (#11202)

Signed-off-by: Jakub Scholz <www@scholzj.com>

see-quick committed:
[system test] Fix downgrade procedure (#11154)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

tinaselenge committed:
Refactor MirrorMaker2Connectors configuration setup (#11150)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

PaulRMellor committed:
docs(kraft): kRaft diagram updates (#11225)

Signed-off-by: prmellor <pmellor@redhat.com>

fvaleri committed:
Adopt new Kafka Connect health check endpoint (#11218)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

PaulRMellor committed:
docs(cruise control): update to cruise control flow diagram (#11226)

Signed-off-by: prmellor <pmellor@redhat.com>

katheris committed:
User Operator handles the cluster CA secret being missing (#11229)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

im-konge committed:
[ST] Add Service to LogCollector's list of resources to collect (#11235)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

im-konge committed:
Add Drain Cleaner 1.3.0 to the Operators repository (#11234)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

im-konge committed:
[ST] Correct names in loggers for KAO ST (#11238)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

im-konge committed:
[ST] Use `AutoOffsetResetStrategy` in clients instead of `OffsetResetStrategy` (#11239)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>

PaulRMellor committed:
docs(downgrade): updates from review of downgrade content (#11231)

Signed-off-by: prmellor <pmellor@redhat.com>

fvaleri committed:
Unify configuration hash annotations (#11237)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

fvaleri committed:
Topic Operator test refactoring (#11220)

Signed-off-by: Federico Valeri <fedevaleri@gmail.com>

ppatierno committed:
Minor refactoring of starting SPS controller method (#11255)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

see-quick committed:
setup.kind fix IPv6/dual provisioning (#11257)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

see-quick committed:
Allow run LoadBalancer tests on TestingFarm (#11222)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

darwin67 committed:
Remove preferIPv4Stack option and let it use JVM defaults (#11247)

Signed-off-by: Darwin D Wu <darwin67@users.noreply.github.com>
Co-authored-by: Darwin D Wu <darwin67@users.noreply.github.com>

see-quick committed:
[system test] [perf] Fix executor handling (#11270)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

PaulRMellor committed:
[DOC] clarifies configuration of jmxOptions and Prometheus Exporter (#11264)

Signed-off-by: prmellor <pmellor@redhat.com>

ppatierno committed:
Updated to Kafka 4.0.0 (#11233)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

katheris committed:
Remove unnecessary executeBlocking call (#11269)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

shk3 committed:
Set units and chart descriptions for MirrorMaker2 Grafana example dashboard (#11263)

Signed-off-by: George H Sun <1467726+shk3@users.noreply.github.com>

ppatierno committed:
Replaced openssl with Java security to add key/cert into keystore (#11224)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

PaulRMellor committed:
docs(logging): log4j2 support updates (#11254)

Signed-off-by: prmellor <pmellor@redhat.com>

PaulRMellor committed:
docs(upgrade): clarifies how upgrade/downgrade paths work (#11290)

Signed-off-by: prmellor <pmellor@redhat.com>

see-quick committed:
Strimzi test container 0.110.0  (#11287)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

showuon committed:
bump the tiered storage plugin version (#11300)

Signed-off-by: Luke Chen <showuon@gmail.com>

see-quick committed:
[system test] [perf] fix 409 error code (#11308)

Signed-off-by: see-quick <maros.orsak159@gmail.com>

Wilya3 committed:
Fix default value typo in README.md (#11314)

Signed-off-by: Wilya3 <77576153+Wilya3@users.noreply.github.com>

Frawless committed:
Bump skodjob-test-docs generator to version 0.4.0 (#11322)

Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

katheris committed:
Store CA generation in Ca class (#11277)

Signed-off-by: Katherine Stanley <11195226+katheris@users.noreply.github.com>

ppatierno committed:
Bumped Vert.x 4.5.14 (#11324)

Signed-off-by: Paolo Patierno <ppatierno@live.com>

PaulRMellor committed:
docs(connect): updates the deployment instructions for kafka connect (#11301)

Signed-off-by: prmellor <pmellor@redhat.com>

ShubhamRwt committed:
Bump the Cruise Control version to 2.5.142 (#11288)

Signed-off-by: ShubhamRwt <shubhamrwt02@gmail.com>

tinaselenge committed:
Mount OAuth related client configuration via volume within Kafka Bridge (#11253)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

tinaselenge committed:
Mount OAuth related client configuration via volume within Kafka Connect (#11306)

Signed-off-by: Gantigmaa Selenge <tina.selenge@gmail.com>

im-konge committed:
Normalize volume size for PVC to skip unnecessary patches (#11315)

pboers1988 committed:
Update ADOPTERS.md add SURF (#11337)

Signed-off-by: Peter Boers <peter.boers@surf.nl>

im-konge committed:
Promote `ContinueReconciliationOnManualRollingUpdateFailure` FG to GA (#11323)

Signed-off-by: Lukas Kral <lukywill16@gmail.com>
Signed-off-by: Lukáš Král <53821852+im-konge@users.noreply.github.com>
Co-authored-by: PaulRMellor <47596553+PaulRMellor@users.noreply.github.com>

im-konge committed:
Clean Azure agent in AZPs before running tests (#11330)

PaulRMellor committed:
docs(opa): update of metrics and security content (#11307)

Signed-off-by: prmellor <pmellor@redhat.com>
